modules/standard/Curl.chpl

```diff
@@ -17,24 +17,326 @@
  * limitations under the License.
  */
 
-/* Libcurl support 
+/*
+
+This module provides support for libcurl, enabling Chapel programs
+to work with many network protocols. 
+
+The `curl homepage <http://curl.haxx.se/libcurl/>`_ describes libcurl thus::
+
+ libcurl is a free and easy-to-use client-side URL transfer library, supporting
+ DICT, FILE, FTP, FTPS, Gopher, HTTP, HTTPS, IMAP, IMAPS, LDAP, LDAPS, POP3,
+ POP3S, RTMP, RTSP, SCP, SFTP, SMTP, SMTPS, Telnet and TFTP.  libcurl supports
+ SSL certificates, HTTP POST, HTTP PUT, FTP uploading, HTTP form based upload,
+ proxies, cookies, user+password authentication (Basic, Digest, NTLM,
+ Negotiate, Kerberos), file transfer resume, http proxy tunneling and more! 
+
+Dependencies
+------------
+
+The Curl functionality in Chapel is dependent on libcurl. For information on
+how to install libcurl, see the
+`curl installation instructions <http://curl.haxx.se/docs/install.html>`_
+
+The environment variables CHPL_AUXIO_INCLUDE and CHPL_AUXIO_LIBS must be set to
+point to the include and lib directories for libcurl respectively. More
+information on these variables can be found in README.auxIO
```

Comment:
Summary from the chat discussion: not for this release, but hopefully for the next one.